pandas/core/indexes/multi.py

```diff
@@ -242,11 +242,35 @@ def __new__(cls, levels=None, codes=None, sortorder=None, names=None,
             result.sortorder = sortorder
 
         if verify_integrity:
-            result._verify_integrity()
+            new_codes = result._verify_integrity()
+            result._codes = new_codes
+
         if _set_identity:
             result._reset_identity()
+
         return result
 
+    def _validate_codes(self, level, code):
```

Comment:
can you try to type annotate level and code here

Reply:
I am not sure about type annotation, could you please point me to any example in the code base?

Reply:
sure we have started generic codes in pandas._typing (see how they are used); this can be a followup